src/main/java/io/mosip/iiitb/odte/utils/RSACryptoTool.java

```diff
@@ -0,0 +1,110 @@
+package io.mosip.iiitb.odte.utils;
+
+import com.google.inject.Inject;
+import io.mosip.iiitb.odte.config.OnDemandAppConfig;
+import org.slf4j.Logger;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.OAEPParameterSpec;
+import javax.crypto.spec.PSource;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.*;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.MGF1ParameterSpec;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+
+public class RSACryptoTool {
+    final PrivateKey pk;
+    Cipher cipher = null;
+    final Logger logger;
+
+    @Inject
+    public RSACryptoTool(
+        OnDemandAppConfig onDemandAppConfig,
+        Logger logger
+    ) throws NoSuchAlgorithmException, InvalidKeySpecException, IOException {
+        this.logger = logger;
+        String privateKeyFileLocation = onDemandAppConfig.privateKeyFileLocation();
+        this.pk = loadPrivateKey(privateKeyFileLocation);
+        try {
+            Cipher cipher = Cipher.getInstance("RSA/ECB/OAEPWithSHA-256AndMGF1Padding");
+            OAEPParameterSpec oaepParams = new OAEPParameterSpec("SHA-256", "MGF1", MGF1ParameterSpec.SHA256, PSource.PSpecified.DEFAULT);
+            cipher.init(Cipher.DECRYPT_MODE, this.pk, oaepParams);
+            this.cipher = cipher;
+        } catch (NoSuchAlgorithmException ex) {
+            logger.error(ex.getMessage());
+            System.exit(11);
+        } catch (NoSuchPaddingException ex) {
+            logger.error(ex.getMessage());
+            System.exit(12);
+        } catch (InvalidKeyException ex) {
+            logger.error(ex.getMessage());
+            System.exit(13);
+        } catch (InvalidAlgorithmParameterException ex) {
+            logger.error(ex.getMessage());
+            System.exit(14);
+        }
+    }
+    public byte[] decryptData(byte[] encryptedData) throws Exception {
+        return this.cipher.doFinal(encryptedData);
```

Comment:
Not sure if this will be thread safe

Reply:
Simple solution I can think on top of my head is to create new Cipher instance and then use that to decrypt(like the implementation in [encryptData](https://github.com/abdulbathish/jackfruit/blob/e070c2dbbaa1362f04b818ad1fc664f7fdaffe50/src/main/java/io/mosip/iiitb/odte/utils/RSACryptoTool.java#L65). 
If you got any other recommend approaches to this problem,  please suggest them

Reply:
Decided to Create a new Cipher Instance on every decryption, when the program initially starts a sanity check is done that cipher instance can be initialized and right config values are passed